wnfs/src/private/node.rs

```diff
@@ -451,14 +445,65 @@ impl PrivateNode {
             .collect())
     }
 
-    pub(crate) async fn load(
+    /// Tries to deserialize and decrypt a PrivateNode at provided PrivateRef.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::rc::Rc;
+    /// use chrono::Utc;
+    /// use rand::thread_rng;
+    /// use wnfs::{
+    ///     private::{PrivateForest, PrivateRef}, PrivateNode,
+    ///     BlockStore, MemoryBlockStore, Namefilter, PrivateDirectory, PrivateOpResult,
+    /// };
+    ///
+    /// #[async_std::main]
+    /// async fn main() {
+    ///     let store = &mut MemoryBlockStore::default();
+    ///     let rng = &mut thread_rng();
+    ///     let forest = &mut Rc::new(PrivateForest::new());
+    ///     let dir = Rc::new(PrivateDirectory::new(
+    ///         Namefilter::default(),
+    ///         Utc::now(),
+    ///         rng,
+    ///     ));
+    ///
+    ///     let node = PrivateNode::Dir(dir);
+    ///
+    ///     let private_ref = node.store(forest, store, rng).await.unwrap();
+    ///
+    ///     assert_eq!(
+    ///         PrivateNode::load(&private_ref, forest, store).await.unwrap(),
+    ///         node
+    ///     );
+    /// }
+    /// ```
+    pub async fn load(
         private_ref: &PrivateRef,
+        forest: &PrivateForest,
+        store: &impl BlockStore,
+    ) -> Result<PrivateNode> {
+        let cid = match forest
+            .get_encrypted(&private_ref.saturated_name_hash, store)
+            .await?
+        {
+            Some(cids) if cids.contains(&private_ref.content_cid) => private_ref.content_cid,
+            _ => return Err(FsError::NotFound.into()),
+        };
+
+        Self::from_cid(cid, &private_ref.temporal_key, store).await
+    }
+
+    pub(crate) async fn from_cid(
```

Comment:
I renamed the old `PrivateNode::load` into `from_cid` essentially.
Other than that it doesn't take the whole `PrivateRef` anymore, but just the `Cid` and the `TemporalKey`.
That felt like it made sense, since it doesn't do any `PrivateForest` lookups anyway. (now explicitly!)